editor/editor/mode/visual.js

```diff
@@ -1,13 +1,22 @@
 /**
+ * External dependencies
+ */
+import { findIndex } from 'lodash';
+
+/**
  * Internal dependencies
  */
 import InserterButton from '../../inserter/button';
 
 function Blocks( { blocks, onChange } ) {
-	const onChangeBlock = ( index ) => ( changes ) => {
+	const onChangeBlock = ( uid ) => ( changes ) => {
+		const index = findIndex( blocks, { uid } );
 		const newBlock = {
```

Comment:
Not really, If I call `onChange( { myAttribute: { key1: '1', key3: '2' } } )`  and I have `{ myAttribute: { key2: 2 } }` in the current state. I don't want to keep the `key2`. I want to replace the attribute completely.